### ISSUE TITLE
Default session privacy setting

### ISSUE THREAD

**Issue by Lualttt** [Yellow-Dog-Man/Resonite-Issues#2253]
### Is your feature request related to a problem? Please describe.

I personally only really start private or contacts sessions but it's quite annoying to make a custom session and then going to settings and then pressing on what privacy I want.

### Describe the solution you'd like

Having a default session privacy setting in the settings menu in the dashboard would make this easier.

### Describe alternatives you've considered

Manually setting it everytime.

### Requesters

discord: lualt `538770131359236105`

**Comment by Lualttt** [comment]
I didn't know about the invisible status defaulting to private, because of that I agree that it would be useful to be able to set the default for every status

**Comment by Frooxius** [comment]
We could add this. The main question is, do we want to allow to override this for every single status? Or just Invisible & non-Invisible?

I can see people wanting to do different ones for Busy, Online and Sociable maybe, so there's a benefit there. But I don't also want to clutter the settings too much and make them too overwhelming.

**Comment by shiftyscales** [comment]
For my own personal use, I wouldn't need the split at all as I'd probably always want my sessions to always default to open hidden/private- as I just raise the visibility as-needed once the session is already open.

I see a lot more potential for user error by adding the extra granularity.

E.g. a user has a hidden project/world they want to work on/view, opens the world- but forgot they were in the 'sociable' status, and now they have a public instance of their private work.

I haven't tested it in a while but IIRC the current behaviour is that worlds will remember the last value they were set to- at least with regards to the 'hidden' property- so it'd be preferable at least that if a world is saved/defaulted to hidden it would remain that way- and the setting would be more of a 'suggestion' than a rule in those instances.

E.g. if it favoured making a world more private when possible, but not less- particularly since the user can still explicitly set the session visibility once the world is open.

**Comment by shadowpanther** [comment]
> We could add this. The main question is, do we want to allow to override this for every single status? Or just Invisible & non-Invisible?

**Invisible** already has a **Private** session by default, so it probably could stay like that. I don't see any purpose for the **Invisible** user to want anything else by default.
**Busy**, **Online** and **Sociable** seem like a good set of statuses to add settings for, as these indicate different level of wanting to be around other people.
**Away** probably shouldn't have such a setting. And I feel like it shouldn't even be a manually-set status, but that's for another issue.

> But I don't also want to clutter the settings too much and make them too overwhelming.

It could be a sub-category, like the ones for Relay priority or Hosts permissions. Then you'd only have one settings item for the main list.

**Comment by XDelta** [comment]
My own thoughts would be that per status would be too much and unneeded for how hard it is to just change the visibility normally. Only probably need a single override for the visible statuses together replacing the default behavior. 

Invisible doesn't need it in my mind as you'll always need to invite initially anyways regardless on change to be visible.

Away wouldn't need it and is usually auto set anyway. Most of the time, it would be automatically swapping back to online anyway.

Maybe some argument for sociable to have it so you can set it as contacts+ or anyone if your other online statuses are contacts only or private. 

At least of what I've seen more often is people want to change the default to just not be set as 'anyone', instead rather private or contacts(+) and then change from there later on.

**Comment by Lualttt** [comment]
Yeah personally I like making it private first and making it open when it needs to be.

**Comment by Ardescar** [comment]
I really like to have a setting to open a world always in private.
And then to make the extra step more to open it to ppl when i want it.

It is just not clear that a world from the inventory just goes to "anyone" even if your in DND/busy mode.

Please do something about not defaulting to "anyone" - i had several issues with that allready. 

And its maybe for new users confusing.
